bibliopixel/threads/animation_threading.py

```diff
@@ -18,7 +18,7 @@ def __init__(self, runner, run):
     def stop_thread(self, wait=False):
         if self.thread:
             self.stop_event.set()
-            if wait:
+            if wait and (threading.current_thread() != self.thread):
                 self.thread.join()
```

Comment:
This got closed, but since I wrote this:

Very very very minor quibble:  I'd use `is not`, because it's a tiny bit faster and actually says what you mean.  I don't even know what it means if two threads are "equal", but I certainly know what `is` is. 

It's a tiny bit faster, because `is` or `is not` are always the fastest comparisons - because there is no "magic method" lookup - Python looks to see if the two objects are the same, done!  

If you use `==` or `!=` then Python has to look up a magic method like `__eq__`, then call that method - whose first action is nearly always to check using `is` and short-circuit if `is` is `True`.

Takeaway - if you have a choice between `is`/is not` and `==`/`!=`, always use `is`.

These days, I might even write:

    wait and (threading.current_thread() is not self.thread) and self.thread.join()